rclcpp/test/benchmark/benchmark_client.cpp

```diff
@@ -95,37 +91,21 @@ BENCHMARK_F(ClientPerformanceTest, destroy_client_empty_srv)(benchmark::State &
   outer_client.reset();
 
   reset_heap_counters();
+  std::shared_ptr<rclcpp::Client<test_msgs::srv::Empty>> client(nullptr);
   for (auto _ : state) {
-    state.PauseTiming();
-    auto client = node->create_client<test_msgs::srv::Empty>(empty_service_name);
-    state.ResumeTiming();
+    PERFORMANCE_TEST_FIXTURE_PAUSE_MEASUREMENTS(
+      state,
+    {
+      client = node->create_client<test_msgs::srv::Empty>(empty_service_name);
+    });
+
     benchmark::DoNotOptimize(client);
     benchmark::ClobberMemory();
 
     client.reset();
   }
 }
 
-BENCHMARK_F(ClientPerformanceTest, wait_for_service)(benchmark::State & state) {
```

Comment:
This benchmark was removed because it didn't actually pause timing around the destructors, which turned out to be the majority of the reported time for this benchmark. After I tried that, `wait_for_service` took only 4us, which was a small fraction of the actual runtime, leading this benchmark to run for a couple of minutes.

Reply:
I really think we should try to find a way to address this. Not here, but at some point. It's come up in multiple benchmark PRs now, and there are some benchmarks that we have already committed that suffer from this issue, just not bad enough to remove the test.

I understand what's happening, but I'm afraid that we'll need changes to benchmark itself to fix it...

Maybe we should open an issue on one of our repos to continue discussing possible mitigations.

Reply:
Opened https://github.com/ros2/performance_test_fixture/issues/12 for discussion